src/test/java/co/rsk/federate/signing/hsm/client/PowHSMSigningClientBtcTest.java

```diff
@@ -423,36 +422,41 @@ private void addCommonPegoutLogs(List<LogInfo> logs, BtcTransaction pegoutBtcTx)
 
     private BtcTransaction createPegout(List<Coin> outpointValues,
         List<Address> destinationAddresses, boolean segwit) {
-        BtcTransaction fundingTransaction = new BtcTransaction(btcMainnetParams);
-        fundingTransaction.addInput(BitcoinTestUtils.createHash(1), FIRST_OUTPUT_INDEX,
-            new Script(new byte[]{}));
 
-        for (Coin outpointValue : outpointValues) {
-            fundingTransaction.addOutput(outpointValue, oldFederation.getAddress());
-        }
-
-        BtcTransaction pegoutBtcTx = new BtcTransaction(btcMainnetParams);
+        BtcTransactionBuilder btcTransactionBuilder = new BtcTransactionBuilder();
 
+        // TODO: improve this method to create a more realistic btc segwit transaction
+        //  once {@link SignerMessageBuilder#getSigHashByInputIndex(int)} is refactored to support segwit
         Script inputScriptThatSpendsFromTheFederation = createBaseInputScriptThatSpendsFromTheFederation(
             oldFederation);
 
         Coin fee = Coin.MILLICOIN;
-        for (int outputIndex = 0; outputIndex < outpointValues.size(); outputIndex++) {
-            TransactionInput addedInput = pegoutBtcTx.addInput(
-                fundingTransaction.getOutput(outputIndex));
-
-            addedInput.setScriptSig(inputScriptThatSpendsFromTheFederation);
-
+        for (int inputIndex = 0; inputIndex < outpointValues.size(); inputIndex++) {
+            Coin outpointValue = outpointValues.get(inputIndex);
+            Coin amountToSend = outpointValue.minus(fee);
+            // Iterate over the addresses using inputIndex % addresses.size() to have outputs to different addresses
+            Address destinationAddress = destinationAddresses.get(
+                inputIndex % destinationAddresses.size());
+
+            btcTransactionBuilder
+                .withInput(
+                    btcTransactionBuilder.new InputBuilder(outpointValue).withOutpointIndex(
+                            inputIndex)
+                        .withScriptSig(inputScriptThatSpendsFromTheFederation).build()
+                )
```

Comment:
wdyt about creating the input outside? So the code is easier to follow:

```
inputBuilder = btcTransactionBuilder.new InputBuilder(outpointValue)
input = inputBuilder
             .withOutpointIndex(inputIndex)
             .withScriptSig(inputScriptThatSpendsFromTheFederation)
             .build()

btcTransactionBuilder
             .withInput(input)
```

Reply:
Done